### PR TITLE
Add k3s integrate with harvester document

### PR DESCRIPTION
**Pull request by FrankYang0529**
* Create k3s with harvester node driver.
* Integrate k3s with harvester csi driver and cloud provider.

Related issues:
* https://github.com/harvester/harvester/issues/1812
* https://github.com/harvester/harvester/issues/1814

### CODE REVIEW

File: docs/rancher/k3s-cluster.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Creating an K3s Kubernetes Cluster
+
+Users can now provision K3s Kubernetes clusters on top of the Harvester cluster in Rancher `v2.6.3+` using the built-in Harvester node driver.
+
+![k3s-cluster](assets/rke2-k3s-node-driver.png)
+
+!!! note
+    - Harvester K3s node driver is in tech preview.
+    - VLAN network is required for Harvester node driver.
```
Comment (nunix) [inline docs/rancher/k3s-cluster.md:9]:
```suggestion
    - [VLAN network](https://docs.harvesterhci.io/v1.0/networking/harvester-network/#create-a-vlan-network) is required for Harvester node driver.
```

File: docs/rancher/k3s-cluster.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+4. Click **Harvester**.
+5. Enter your cloud credential name
+6. Select "Imported Harvester" or "External Harvester".
+7. Click **Create**.
+
+![create-harvester-cloud-credentials](assets/create-cloud-credentials.png)
+
+###  Create K3s Kubernetes Cluster
+
+Users can create a K3s Kubernetes cluster from the **Cluster Management** page via the K3s node driver.
```
Comment (nunix) [inline docs/rancher/k3s-cluster.md:25]:
```suggestion
You can create a K3s Kubernetes cluster from the **Cluster Management** page via the K3s node driver.
```

File: docs/rancher/k3s-cluster.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+K3s provisioning relies on the `qemu-guest-agent` to get the IP of the virtual machine. However, it may not be feasible to install `qemu-guest-agent` in an air gapped environment.
+
+You can address the installation constraints with the following options:
+
+Option 1. Use a VM image with `qemu-guest-agent` installed.
+
+Option 2. Configure the `cloud init` user data to enable the VMs to install `qemu-guest-agent` via an HTTP(S) proxy.
+
+Example user data in Harvester node template:
```
Comment (nunix) [inline docs/rancher/k3s-cluster.md:51]:
```suggestion
Example of `user data` in Harvester node template:
```

**Review by nunix — COMMENTED** [review]
Rephrased to active voice.

Also saw that RKE2 page is not in active voice, I will adapt it once this PR is approved for an harmonized phrasing.

File: docs/rancher/k3s-cluster.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# Creating an K3s Kubernetes Cluster
+
+Users can now provision K3s Kubernetes clusters on top of the Harvester cluster in Rancher `v2.6.3+` using the built-in Harvester node driver.
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply docs/rancher/k3s-cluster.md]: Updated it. Thank you.

File: docs/rancher/k3s-cluster.md (hunk 1 above)
Comment (FrankYang0529) [reply docs/rancher/k3s-cluster.md]: Updated it. Thank you.

File: docs/rancher/k3s-cluster.md (hunk 2 above)
Comment (FrankYang0529) [reply docs/rancher/k3s-cluster.md]: Updated it. Thank you.

File: docs/rancher/k3s-cluster.md (hunk 3 above)
Comment (FrankYang0529) [reply docs/rancher/k3s-cluster.md]: Updated it. Thank you.

**Review by nunix — APPROVED** [review]

File: docs/rancher/cloud-provider.md [hunk 5]
```diff
@@ -48,6 +48,25 @@ When spinning up an RKE2 cluster using the Harvester node driver, select the `Ha
...
+```
+# depend on kubectl to operate the Harvester cluster
+./deploy/generate_addon.sh <serviceaccount name> <namespace>
+```
+
+- Edit K3s cluster YAML.
+    - Set `disable-cloud-provider: true` to disable default k3s cloud provider.
+    - Add `cloud-provider=external` to use harvester cloud provider.
+
+  ![](assets/edit-k3s-cluster-yaml-for-harvester-cloud-provider.png)
```
Comment (guangbochen) [inline docs/rancher/cloud-provider.md:64]: Will editing the cluster YAML after installation be supported? if not probably better to highlight it.
And the preferred steps will be:
- Choose the Kubernetes version to be k3s and click the `Edit as YAML` button to config the K3s cluster YAML:
image
  - ...
  - ...

File: mkdocs.yml [hunk 6]
```diff
@@ -7,7 +7,7 @@ repo_name: harvester/harvester
 repo_url: https://github.com/harvester/harvester
 edit_uri: https://github.com/harvester/docs/edit/main/docs/
 # Set to false for local docs development
-strict: true
+strict: false
```
Comment (guangbochen) [inline mkdocs.yml:10]: this has to be `true`

File: docs/rancher/csi-driver.md [hunk 7]
```diff
@@ -32,3 +32,16 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
...
+- [Generate addon configuration](https://github.com/harvester/harvester-csi-driver/blob/master/deploy/generate_addon.sh) and put it in K3s VMs `/etc/kubernetes/cloud-config`.
+
+```
+# depend on kubectl to operate the Harvester cluster
+./deploy/generate_addon.sh <serviceaccount name> <namespace>
+```
+
+- Install `Harvester CSI Driver` from the Rancher marketplace.
+
+  ![](assets/install-harvester-csi-driver-in-k3s.png)
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply docs/rancher/csi-driver.md]: Updated it. Thank you.

File: docs/rancher/cloud-provider.md [hunk 8]
```diff
@@ -48,6 +48,25 @@ When spinning up an RKE2 cluster using the Harvester node driver, select the `Ha
 
   ![](assets/rke2-cloud-provider.png)
 
+### Deploying to the K3s Cluster with Harvester Node Driver
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply docs/rancher/cloud-provider.md]: Updated it. Thank you.

File: mkdocs.yml (hunk 6 above)
Comment (FrankYang0529) [reply mkdocs.yml]: Fixed it. Thank you.

File: docs/rancher/cloud-provider.md (hunk 5 above)
Comment (FrankYang0529) [reply docs/rancher/cloud-provider.md]: Yes, we can edit cluster YAML after installation. I update the steps. For the first step, users can click `Edit as YAML` for a new cluster or `Edit YAML` for existing clusters.

**Review by guangbochen — APPROVED** [review]

**Pull request merged**